keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -535,29 +597,89 @@ Disk-pressure eviction:
   - `reason=Evicted`
   - `message=The node was low on resource: ephemeral-storage. (...)`
 - Container status:
-  - `state=Ternminated`
+  - `state=Terminated`
   - `exitCode=137`
   - `reason=Error`
-- Retriable: Unclear, excessive disk usage suggests a bug or misconfiguration.
-  However, a restart on another node may succeed
 
-##### OOM kill
+##### Container memory limit exceeded
+
+Linux:
 
 - Reproduction: We run a job with a pod which attempts to allocate more
   memory than constrained in the container spec by `resources.limits.memory`
-- Comments: handled by kubelet
+- Comments: handled by kubelet in `kubelet/kubelet_pods.go` which merges-in the status update (setting of the reason field) by the container runtime
 - Pod status:
   - status: OOMKilled
   - `phase=Failed`
   - `reason=`
   - `message=`
 - Container status:
-  - `state=Ternminated`
+  - `state=Terminated`
   - `exitCode=137`
   - `reason=OOMKilled`
-- Retriable: Unclear, but if occurs when
-  `resources.requests.memory=resources.limits.memory` it strongly suggests
-  a software bug.
+
+Windows:
+
+- Reproduction: We run a job with a pod which attempts to allocate more
+  memory than constrained in the container spec by `resources.limits.memory`
+- Comments: there is not clear indication that the container failed due to exceeding memory limit
+- Pod status:
+  - status: Error
+  - `phase=Failed`
+  - `reason=`
+  - `message=`
+- Container status:
+  - `state=Terminated`
+  - `exitCode=1`
+  - `reason=Error`
+
+##### Container ephemeral-storage limit exceeded
+
+- Reproduction: We run a job with a pod which attempts to consume more disk space
+  than constrained in the container spec by `resources.limits.ephemeral-storage`
+- Comments: handled by kubelet in `kubelet/eviction/eviction_manager.go`
+- Pod status:
+  - status: Error
+  - `phase=Failed`
+  - `reason=Evicted`
+  - `message=Pod ephemeral local storage usage exceeds the total limit of containers 1Gi.`
+- Container status:
+  - `state=Terminated`
+  - `exitCode=137`
+  - `reason=Error`
+
+##### Graceful node shutdown
+
+- Reproduction: The node needs to be started with positive `shutdownGracePeriod`,
+  for example `shutdownGracePeriod=30s`. We run a job with a long-running pod,
+  then we prepare the node for shutdown by the command:
+  `dbus-send --system --type=signal /org/freedesktop/login1 org.freedesktop.login1.Manager.PrepareForShutdown boolean:"true"`,
+  which similulates graceful nodes shutdown for maintenance purposes.
+- Comments: handled by kubelet in `kubelet/nodeshutdown/nodeshutdown_manager_linux.go`
+- Pod status:
+  - status: Error
+  - `phase=Failed`
+  - `reason=Terminated`
+  - `message=Pod was terminated in response to imminent node shutdown.`
+- Container status:
+  - `state=Terminated`
+  - `exitCode=137`
+  - `reason=Error`
+
+##### Pod admission error
```

Comment:
similar can likely be achieved when static pod that is taking over the leftover of resources is scheduled while job is in process of being scheduled on the node. The message will likely be different

Reply:
indeed, there are a couple of scenarios. Updated the KEP to mention that explicitely.